app/_features/event/components/event-list.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import Header from '@/_shared/components/header/header';
+import { EventType } from '@/_shared/types/event';
+import { Button, Link, Tab, Tabs } from '@nextui-org/react';
+import { useEffect, useState } from 'react';
+import { EventCard } from './event-card';
+import EditIcon from '@/_shared/components/icons/edit-icon';
+import ChevronLeft from '@/_shared/components/icons/chevron-left';
+import ChevronRight from '@/_shared/components/icons/chevron-right';
+
+export default function EventList({
+  events,
+}: {
+  events: EventType.ListEventsResponse;
+}) {
+  const [listEvent, setListEvent] = useState<
+    EventType.ListEventsResponse['data']
+  >([]);
+
+  useEffect(() => {
+    if (events) {
+      setListEvent(events.data);
+    }
+  }, [events]);
```

Comment:
`events` props here should be already passing an array of events regardless it is empty or not. This useEffect and useState are making unnecessary component re-render. Please remove the useEffect and useState.